app/models/legacy/division.rb

```diff
@@ -0,0 +1,50 @@
+# note, it's not ideal to muddy our app/models directory with the migration code,
+# but the rail class loading behavior seemed happiest with this location.
+# can perhaps sort that out later and force to work from a different source directory
+
+module Legacy
+
+  class Division < ActiveRecord::Base
+    establish_connection :legacy
+
+    include LegacyModel
+
+    def migration_data
+      if id == super_division
+        parent_id = ::Division::ROOT_ID
+      else
+        parent_id = super_division
+      end
+      data = {
+          id: id,
+          parent_id: parent_id,
+          name: name,
+          description: description,
+          # note, no well defined division currency in current model (aside from the global AR$ default)
+          # currency_id:
+      }
+      data
+    end
```

Comment:
Why bother instantiating this data var? You could just make the hash the last thing and it will be returned.